test/jdk/com/sun/java/swing/plaf/gtk/TestFileChooserSingleDirectorySelection.java

```diff
@@ -189,30 +184,40 @@ private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
         }
     }
 
-    private static void doTesting(UIManager.LookAndFeelInfo laf, int xOffset) {
-        Point frameLocation = fileChooser.getLocationOnScreen();
-        int frameWidth = frame.getWidth();
-        int frameHeight = frame.getHeight();
-
-        Point btnLocation = getSelectedFilesButton.getLocationOnScreen();
-        int btnWidth = getSelectedFilesButton.getWidth();
-        int btnHeight = getSelectedFilesButton.getHeight();
-        clickMouse(frameLocation, 0, frameHeight, xOffset);
-        clickMouse(btnLocation, btnWidth, btnHeight, 0);
+    private static void doTesting(UIManager.LookAndFeelInfo laf, int xOffset)
+                                    throws Exception {
+        SwingUtilities.invokeAndWait(() -> {
+            Point fileChooserLocation = fileChooser.getLocationOnScreen();
+            fileChooserLocation.y += frame.getHeight() / 3;
+            clickLocation = new Point(fileChooserLocation);
+        });
+        clickMouse(clickLocation, xOffset);
         checkResult(laf);
     }
 
-    private static void clickMouse(Point point, int width, int height,
-                                   int xOffset) {
-        robot.mouseMove(point.x + width/2 + xOffset , point.y + height/3);
+    private static void clickMouse(Point point, int xOffset) {
+        robot.mouseMove(point.x + xOffset , point.y);
         robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
         robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
         robot.delay(100);
+        robot.waitForIdle();
+    }
+
+    private static void checkResult(UIManager.LookAndFeelInfo laf) throws Exception {
```

Comment:
Instead of passing the laf through all the methods, can't we create and use a class variable?

Reply:
> Instead of passing the laf through all the methods, can't we create and use a class variable?

Can be done but can't see any functional improvements. This seems ok to me.

Reply:
Yeah, those are to reduce redundancy in code, not functional...... Otherwise the fix looks fine.

Reply:
It's not redundancy: passing the `laf` explicitly makes the data flow clearer, even though `laf` isn't directly used.

I agree that making `laf` a class field doesn't provide much benefit.

In a way, it could be removed completely, however, reporting the L&F at which the test case failed gives more details, you don't need to look at the test log to find the L&F.